### PR TITLE
Some improvements to EvaluationContext

### PR DESCRIPTION
**Pull request by agentgonzo**
More of a suggestion/discussion than a real PR.

@justinabrahms , let me know your thoughts.

### CODE REVIEW

File: src/main/java/dev/openfeature/javasdk/EvaluationContext.java [hunk 1]
```diff
@@ -103,4 +110,49 @@ public static EvaluationContext merge(EvaluationContext ctx1, EvaluationContext
...
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> (String)e.getValue()))
+        );
+        context.booleanAttributes.putAll(
+            map.entrySet()
+                .stream()
+                .filter(entry -> entry.getValue() instanceof Boolean)
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> (Boolean)e.getValue()))
+        );
+//        Hmmm, this won't work as we already have a string type.
+//        I would recommend changing the type to Map<String, JsonNode> from Jackson
```
Comment (justinabrahms) [inline src/main/java/dev/openfeature/javasdk/EvaluationContext.java:135]: I'm a jackson newbie, but my understanding is that jsonnode's are an immutable json tree structure. That seems fine to me for this.

**Comment by beeme1mr** [comment]
@agentgonzo, this seems like a nice change. Would you be able to get it into a mergeable state?

@justinabrahms any concerns about this?

**Comment by justinabrahms** [comment]
@beeme1mr No strong concerns. There are backwards incompatibilities here, which I don't *love*, but it's still early days so.. seems inevitable?

**Comment by justinabrahms** [comment]
@agentgonzo Do you have an intent to finish up this PR? Or should I close it?

**Comment by beeme1mr** [comment]
Closing due to inactivity.

**Comment by agentgonzo** [comment]
This completely fell off my radar. I'm away for a week, but will try to get a decent mergeable PR ready when I get back

**Comment by beeme1mr** [comment]
@agentgonzo sounds good! Feel free to reopen the PR when it's ready.

**Pull request closed without merging**